fedimint-server/src/consensus/mod.rs

```diff
@@ -142,30 +133,34 @@ impl FedimintConsensus {
                 |dbtx| {
                     let consensus_outcome = consensus_outcome.clone();
                     let reference_rejected_txs = reference_rejected_txs.clone();
-                    let peers: BTreeSet<PeerId> = consensus_outcome.contributions.keys().copied().collect();
 
                     Box::pin(async move {
-                        let epoch = consensus_outcome.epoch;
-                        let outcome = consensus_outcome.clone();
-
-                        let UnzipConsensusItem {
-                            epoch_outcome_signature_share: _epoch_outcome_signature_share_cis,
-                            client_config_signature_share: _client_config_signature_share_cis,
-                            transaction: transaction_cis,
-                            consensus_upgrade: consensus_upgrade_cis,
-                            module: module_cis,
-                        } = consensus_outcome
+                        let mut rejected_txs = BTreeSet::new();
+
+                        let items = consensus_outcome.clone()
                             .contributions
                             .into_iter()
-                            .flat_map(|(peer, cis)| cis.into_iter().map(move |ci| (peer, ci)))
-                            .unzip_consensus_item();
-
-                        self.process_module_consensus_items(dbtx, &module_cis, &peers).await;
-                        self.process_upgrade_items(dbtx, &consensus_upgrade_cis).await;
-
-                        let rejected_txs = self
-                            .process_transactions(dbtx, epoch, &transaction_cis)
-                            .await;
+                            .flat_map(|(peer_id, items)| items.into_iter().map(move |item| (peer_id, item)));
+
+                        for (peer_id, consensus_item) in items {
+                            dbtx.set_tx_savepoint()
+                                .await
+                                .expect("Setting transaction savepoint failed");
+
+                            let result = self.process_consensus_item(dbtx, consensus_item.clone(), peer_id).await;
+
+                            if result.is_err(){
+                                dbtx.rollback_tx_to_savepoint()
+                                    .await
+                                    .expect("Rolling back transaction to savepoint failed");
+
+                                if let ConsensusItem::Transaction(transaction) = consensus_item.clone() {
+                                    if self.process_consensus_item(dbtx, consensus_item, peer_id).await.is_err() {
```

Comment:
And/or this one?

Reply:
This call actually have not been there in the first place.